server/games/ladder_game.py

```diff
@@ -1,23 +1,61 @@
+import asyncio
 import logging
 from typing import Optional
 
 from server.config import config
 from server.players import Player
-from server.rating import RatingType
 
 from .game import Game
 from .game_results import ArmyOutcome, GameOutcome
-from .typedefs import FeaturedModType, GameType, InitMode
+from .typedefs import GameState, GameType, InitMode
 
 logger = logging.getLogger(__name__)
 
 
+class GameClosedError(Exception):
+    """
+    The game has been closed during the setup phase
+    """
+
+    def __init__(self, player: Player, *args, **kwargs):
```

Comment:
Why does it even have `args` and `kwargs` still?

Reply:
because in `BaseException` it looks like this:
```python
    def __init__(self, *args, **kwargs): # real signature unknown
        pass
```
I'm not really sure how to interpret that comment. I just assumed that I would need the two arguments as well.

Reply:
Just remove them. It’s that way in base exception because it doesn’t know what specific function signatures the subclasses will have. But for a custom error you should just define the parameters you need. In this case we just need 1 for player and that’s it.